core/src/main/scala/org/http4s/headers/Authorization.scala

```diff
@@ -7,6 +7,9 @@ import org.http4s.util.Writer
 object Authorization extends HeaderKey.Internal[Authorization] with HeaderKey.Singleton {
   override def parse(s: String): ParseResult[Authorization] =
     HttpHeaderParser.AUTHORIZATION(s)
+
+  def apply(basic: BasicCredentials): Authorization =
```

Comment:
This is still a special case that suggests a typeclass, but `BasicCredentials` is the only one that we treat special.